bundle/config/mutator/resolve_variable_references.go

```diff
@@ -205,3 +227,45 @@ func (m *resolveVariableReferences) resolveOnce(b *bundle.Bundle, prefixes []dyn
 
 	return hasUpdates, diags
 }
+
+// selectivelyMutate applies a function to a subset of the configuration
+func (m *resolveVariableReferences) selectivelyMutate(b *bundle.Bundle, fn func(value dyn.Value) (dyn.Value, error)) error {
+	included := []string{
+		"variables",
+		"bundle",
+		"include",
+		"workspace",
+		"artifacts",
+		"targets",
+		"environments",
+		"sync",
+		"run_as",
+		"presets",
+		"experimental",
+		"permissions",
+	}
+
+	if m.includeResources {
+		included = append(included, "resources")
+	}
+
+	return b.Config.Mutate(func(root dyn.Value) (dyn.Value, error) {
+		includedRoot, err := merge.Select(root, included)
+		if err != nil {
+			return dyn.InvalidValue, err
+		}
+
+		excludedRoot, err := merge.AntiSelect(root, included)
```

Comment:
I don't understand why we selecting included array and then "anti-selecting" included array.

I assume "anti-selecting" means "drop these keys from the map".

Reply:
Right. So it will work in cases where there are several keys that we don't include, this way no keys are going to be lost because they are either in includedRoot or excludedRoot